qiskit/providers/aer/backends/aerbackend.py

```diff
@@ -313,6 +322,217 @@ def _run(self, qobj, job_id=''):
+        body, = inst.params
+
+        self._last_flow_id += 1
+        loop_id = self._last_flow_id
+        loop_name = f'while_{loop_id}'
+
+        ret = QuantumCircuit()
+        for qr in body.qregs:
+            ret.add_register(qr)
+        for cr in body.cregs:
+            ret.add_register(cr)
+        if len(ret.clbits) == 0:
+            ret.add_bits(body.clbits)
+        if len(ret.qubits) == 0:
+            ret.add_bits(body.qubits)
+
+        continue_label = f'{loop_name}_continue'
+        loop_start_label = f'{loop_name}_start'
+        break_label = f'{loop_name}_end'
+        inlined_body = self._inline_circuit(body, continue_label, break_label)
+        ret.append(inlined_body,
+                   range(inlined_body.num_qubits),
+                   range(inlined_body.num_clbits))
+        ret.append(AerJump(continue_label, ret.num_qubits),
+                   range(ret.num_qubits), [])
+        ret.append(AerMark(break_label, inlined_body.num_qubits),
+                   range(inlined_body.num_qubits), [])
+        return condition_tuple, continue_label, loop_start_label, break_label, ret
+
+    def _inline_if_else_op(self, inst, continue_label, break_label):
+        condition_tuple = inst.condition
+        true_body, false_body = inst.params
+
+        self._last_flow_id += 1
+        if_id = self._last_flow_id
+        if_name = f'if_{if_id}'
+
+        ret = QuantumCircuit()
+        for qr in true_body.qregs:
+            ret.add_register(qr)
+        for cr in true_body.cregs:
+            ret.add_register(cr)
+        if len(ret.clbits) == 0:
+            ret.add_bits(true_body.clbits)
+        if len(ret.qubits) == 0:
+            ret.add_bits(true_body.qubits)
+
+        if_true_label = f'{if_name}_true'
+        if_else_label = f'{if_name}_else'
+        if_end_label = f'{if_name}_end'
+
+        ret.append(AerMark(if_true_label, ret.num_qubits),
+                   range(ret.num_qubits), [])
+        ret.append(self._inline_circuit(true_body, continue_label, break_label),
+                   range(ret.num_qubits),
+                   range(ret.num_clbits))
+
+        if false_body:
+            ret.append(AerJump(if_end_label, ret.num_qubits),
+                       range(ret.num_qubits), [])
+
+            ret.append(AerMark(if_else_label, ret.num_qubits),
+                       range(ret.num_qubits), [])
+
+            ret.append(self._inline_circuit(false_body, continue_label, break_label),
+                       range(ret.num_qubits),
+                       range(ret.num_clbits))
+
+        ret.append(AerMark(if_end_label, ret.num_qubits),
+                   range(ret.num_qubits), [])
+
+        return condition_tuple, if_true_label, if_else_label if false_body else if_end_label, ret
+
+    def _convert_c_if_args(self, cond_tuple):
+        return [1 if elem is True else 0 if elem is False else elem for elem in cond_tuple]
+
+    def _inline_circuit(self, circ, continue_label, break_label):
```

Comment:
Can you add doc strings for all these these inline helper functions. They are quite complicated and it is not obvious what they do